include/sick_tim/sick_tim_common_tcp.h

```diff
@@ -48,7 +48,7 @@ namespace sick_tim
 class SickTimCommonTcp : public SickTimCommon
 {
 public:
-  SickTimCommonTcp(const std::string &hostname, const std::string &port, AbstractParser* parser);
+  SickTimCommonTcp(const std::string &hostname, const std::string &port, const std::string &timelimit, AbstractParser* parser);
```

Comment:
`timelimit` should be an int, not a string (don't be fooled by `port`; that's a string because `tcp::resolver::query` wants a string there). Also on the parameter server.